api/axis.py

```diff
@@ -323,36 +323,34 @@ def request(
         headers = dict(extra_headers)
         headers["Authorization"] = "Bearer " + self.token
         headers["Library"] = self.library_id
-        if exception_on_401:
-            disallowed_response_codes = ["401"]
-        else:
-            disallowed_response_codes = None
         response = self._make_request(
             url=url,
             method=method,
             headers=headers,
             data=data,
             params=params,
-            disallowed_response_codes=disallowed_response_codes,
             **kwargs,
         )
-        if response.status_code == 401:
-            # This must be our first 401, since our second 401 will
-            # make _make_request raise a RemoteIntegrationException.
-            #
-            # The token has expired. Get a new token and try again.
-            self.token = None
-            return self.request(
-                url=url,
-                method=method,
-                extra_headers=extra_headers,
-                data=data,
-                params=params,
-                exception_on_401=True,
-                **kwargs,
-            )
-        else:
-            return response
+        if response.status_code == 401 and not request_retried:
+            parsed = StatusResponseParser().process_first(response.content)
+            if parsed is None or parsed[0] in [1001, 1002, 1003]:
+                # The token is probably expired. Get a new token and try again.
+                # Axis 360's status codes mean:
+                #   1001: Invalid token
+                #   1002: Token expired
+                #   1003: Token is invalid for given library
```

Comment:
Wondering if we should retry for `1003`, since it might mean retrying more than we need to. But it's probably 50/50 at worst, so maybe worth attempting anyway.

Reply:
I wasn't sure on that one. I'll pull it out and we can add it back if needed.